src/librados/librados_c.cc

```diff
@@ -3024,6 +3024,7 @@ extern "C" int _rados_aio_unlock(rados_ioctx_t io, const char *o, const char *na
   librados::IoCtx ctx;
   librados::IoCtx::from_rados_ioctx_t(io, ctx);
   librados::AioCompletionImpl *comp = (librados::AioCompletionImpl*)completion;
+  comp->get();
```

Comment:
we decrease the refcount of inner impl pointer in destructor of `AioCompletion`, so we should hold a strong reference .